Remove balanced attached node volumes

ravisantoshgudimetla opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?
<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature
/kind design

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

/kind cleanup

#### What this PR does / why we need it:
https://github.com/kubernetes/kubernetes/pull/60525/ introduced
Balanced attached node volumes feature gate to include volume
count for prioritizing nodes. The reason for introducing this
flag was its usefulness in Red Hat OpenShift Online environment
which is not being used any more. So, removing the flag
as it helps in maintainability of the scheduler code base
as mentioned at https://github.com/kubernetes/kubernetes/pull/101489#discussion_r633568987
#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
Remove balanced attached node volumes feature
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs
Remove balanced attached node volumes feature
```

ravisantoshgudimetla commented:
/sig scheduling

ravisantoshgudimetla commented:
/retest

ingvagabund commented on pkg/scheduler/framework/interface.go:
```diff
@@ -478,7 +478,7 @@ type Framework interface {
 	// removed from it to evaluate the possibility of preempting them to
 	// schedule the target pod.
-	RunFilterPlugins(ctx context.Context, state *CycleState, pod *v1.Pod, nodeInfo *NodeInfo) PluginToStatus
+	//RunFilterPlugins(ctx context.Context, state *CycleState, pod *v1.Pod, nodeInfo *NodeInfo) PluginToStatus
```
Can you move removal of the methods into a separate PR? Building your changes locally without changing the `Framework` interface, I don't see any compilation errors.

alculquicondor replied:
Are these comments intentional?

ravisantoshgudimetla replied:
I see the following error in my GoLand IDE. While this is not causing compilation issues, I see the following:

```
Duplicate method 'RunFilterPlugins'
Duplicate method 'RunPreFilterExtensionAddPod'
Duplicate method 'RunPreFilterExtensionRemovePod'
Duplicate method 'RunPreScorePlugins'
Duplicate method 'RunScorePlugins'
```

coming from ```Handle -> PluginsRunner```
I was initially thinking it was caused by my changes but seems master is also having the same problem. Unrelated but why to have those methods implicitly and explicitly within embedded interfaces and Framework interface?

alculquicondor replied:
Ah... it seems like a leftover from a refactoring. I'll open a separate PR for that.

ingvagabund reviewed (changes requested):
Mostly removing all the blocks which are conditioned by the feature being enabled. Do you also plan to remove the feature flag itself? It's still alpha. No guarantees the flag will be there. If the featuregate for `BalanceAttachedNodeVolumes` is set but the name is not recognized, will the scheduler complain?

alculquicondor reviewed (commented):
+1 to remove the feature gate. The scheduler won't start if the feature name is not recognized, but that's expected. So the feature name should be mentioned in the release note (and perhaps add ACTION REQUIRED).

ahg-g commented:
Please also remove the feature from [kube_features.go](https://github.com/kubernetes/kubernetes/blob/f31bfd8355cc1fb9cf0cb78a24f94d6d7a9012ea/pkg/features/kube_features.go#L220); and send a followup PR to cleanup the docs as well: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/

ravisantoshgudimetla commented:
> Please also remove the feature from kube_features.go; and send a followup PR to cleanup the docs as well: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/

Sure, I thought I pushed it the modified kube_features.go as well but missed it. I've pushed it now. I'll send a docs PR soon.

ravisantoshgudimetla commented:
I think I should send an email to devel mailing list to make sure that this comes as not a surprise to users again later.

ahg-g commented:
> I think I should send an email to devel mailing list to make sure that this comes as not a surprise to users again later.

sounds good, pls cc sig-scheduling mailing list as well.

alculquicondor commented on pkg/scheduler/framework/plugins/noderesources/balanced_allocation_test.go:
```diff
@@ -24,114 +24,14 @@ import (
-		},
-		NodeName: "machine4",
-	}
+
```
nit: remove empty line

ravisantoshgudimetla replied:
Done. Let's wait for a week more and get this feature removed.

alculquicondor replied:
Let's proceed?

ravisantoshgudimetla replied:
+1

alculquicondor reviewed (commented):
Anything else you plan to do or can you remove the WIP from the name?

alculquicondor commented:
This needs another rebase :(

ravisantoshgudimetla committed:
Remove balanced attached node volumes

kubernetes#60525 introduced
Balanced attached node volumes feature gate to include volume
count for prioritizing nodes. The reason for introducing this
flag was its usefulness in Red Hat OpenShift Online environment
which is not being used any more. So, removing the flag
as it helps in maintainability of the scheduler code base
as mentioned at kubernetes#101489 (comment)

alculquicondor reviewed (commented):
/triage accepted
/lgtm
/approve

alculquicondor commented:
/priority important-soon

ahg-g commented:
/lgtm
/approve
